client.go

```diff
@@ -320,11 +325,171 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return block.Offsets[0], nil
 }
 
+func (client *client) CommitOffset(consumerGroup string, topic string, partitionID int32, offset int64, metadata string) error {
+	return client.tryCommitOffset(consumerGroup, topic, partitionID, offset, metadata, 5)
+}
+
+func (client *client) tryCommitOffset(consumerGroup string, topic string, partitionID int32, offset int64, metadata string, attemptsRemaining int) error {
```

Comment:
re. group vs id etc, I think `Consumer.Group` should be a config, and these should be methods on `PartitionConsumer` that use that config etc.